Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -2193,11 +2194,37 @@ - (void)resetFrame {
 - (void)imagePickerController:(UIImagePickerController *)picker
     didFinishPickingMediaWithInfo:(NSDictionary<NSString *, id> *)info
 {
-    OWSAssert([NSThread isMainThread]);
-    
     [UIUtil modalCompletionBlock]();
     [self resetFrame];
 
+    NSURL *referenceURL = [info valueForKey:UIImagePickerControllerReferenceURL];
+    if (!referenceURL) {
+        DDLogError(@"Could not retrieve reference URL for picked asset");
+        OWSAssert(0);
```

Comment:
I hit this assertion failure on iOS9.3 (haven't tested on 10).

**steps:**

- Go to thread
- Paperclip
- Camera
- take a photo 
- use photo